great_expectations/core/expectation_diagnostics/expectation_diagnostics.py

```diff
@@ -215,3 +304,86 @@ def _convert_checks_into_output_message(
+
+    @staticmethod
+    def _check_input_validation(
+        expectation_instance,
+        examples: List[ExpectationTestDataCases],
+    ) -> ExpectationDiagnosticCheckMessage:
+        """Check that the validate_configuration method returns True"""
+        passed = False
+        sub_messages = []
+        try:
+            first_test = examples[0]["tests"][0]
+        except IndexError:
+            sub_messages.append(
+                {
+                    "message": "No example found to get kwargs for ExpectationConfiguration",
+                    "passed": False,
+                }
+            )
+        else:
+            expectation_config = ExpectationConfiguration(
+                expectation_type=expectation_instance.expectation_type,
+                kwargs=first_test.input,
+            )
+            passed = expectation_instance.validate_configuration(expectation_config)
+
+        return ExpectationDiagnosticCheckMessage(
+            message="Has basic input validation and type checking",
+            passed=passed,
+            sub_messages=sub_messages,
+        )
+
+    @staticmethod
+    def _check_renderer_methods(
+        expectation_instance,
+    ) -> ExpectationDiagnosticCheckMessage:
+        """Check if all statment renderers are defined"""
+        passed = False
+        # For now, don't include the "question" and "descriptive" types since they are so
+        # sparsely implemented
+        # all_renderer_types = {"diagnostic", "prescriptive", "question", "descriptive"}
+        all_renderer_types = {"diagnostic", "prescriptive"}
+        renderer_names = [
+            name
+            for name in dir(expectation_instance)
+            if name.endswith("renderer") and name.startswith("_")
+        ]
+        renderer_types = {name.split("_")[1] for name in renderer_names}
+        if renderer_types - {"question", "descriptive"} == all_renderer_types:
+            passed = True
+        return ExpectationDiagnosticCheckMessage(
+            # message="Has all four statement Renderers: question, descriptive, prescriptive, diagnostic",
+            message="Has both statement Renderers: prescriptive and diagnostic",
+            passed=passed,
+        )
+
+    @staticmethod
+    def _check_linting(expectation_instance) -> ExpectationDiagnosticCheckMessage:
+        """Check if linting checks pass for Expectation"""
+        # TODO: Perform linting checks instead of just giving thumbs up
+        return ExpectationDiagnosticCheckMessage(
+            message="Passes all linting checks",
+            passed=True,
+        )
+
+    @staticmethod
+    def _check_full_test_suite(
+        library_metadata: AugmentedLibraryMetadata,
+    ) -> ExpectationDiagnosticCheckMessage:
+        """Check library_metadata to see if Expectation has a full test suite"""
+        return ExpectationDiagnosticCheckMessage(
+            message="Has a full suite of tests, as determined by project code standards",
```

Comment:
I think we may want to change this terminology to something like "as determined by a code owner" given that this is a manual check and not able to be fulfilled by implementing from a guide.